device/channels.go

```diff
@@ -19,13 +19,13 @@ import (
 // call wg.Done to remove the initial reference.
 // When the refcount hits 0, the queue's channel is closed.
 type outboundQueue struct {
-	c  chan *QueueOutboundElement
+	c  chan *[]*QueueOutboundElement
```

Comment:
why pointer to slice out of curiosity? seems weird enough to be worth calling out somewhere to explain what's happening.

Reply:
We were pooling the slice headers as well as the content, but we since moved to a container in order to shard a mutex, which is also cleaner for this reason.